mplc/splitter.py

```diff
@@ -89,6 +82,69 @@ def copy(self):
         return self.__copy__()
 
 
+class FlexibleSplitter(Splitter):
+    name = 'Fully flexible samples split'
+
+    def __init__(self, amounts_per_partner, configuration, **kwargs):
+
+        logger.info("Proceeding to a flexible split as requested. Please note that the flexible "
+                    "split currently discards the amounts_per_partner (if provided) and infers amounts of samples "
+                    "per partner from the samples_split_configuration provided.")
+
+        # First we re-assemble the split configuration per cluster, and check its coherence
+        self.samples_split_grouped_by_cluster = list(zip(*configuration))
+        for idx, cluster in enumerate(self.samples_split_grouped_by_cluster):
+            if np.sum(cluster) > 1:
+                raise ValueError(f"Amounts of samples of class {idx} split among partners exceed 100%, "
+                                 f"the dataset split cannot be performed.")
+
+        # Init of the superclass to inherit its methods
+        super().__init__(amounts_per_partner, **kwargs)
+
+    def __str__(self):
+        return self.name + str(list(zip(self.samples_split_grouped_by_cluster)))
```

Comment:
This could be re-write with f-string. Moreover, isn't it long ? lists of lists are not really easy to read.

Reply:
Yes, I'll remove that. I copied blindly on the `Advanced Splitter`, which `__str__` method is rewritten to include how it is configured. It doesn't seem that useful actually